spec/models/submission_status_giver_spec.rb

```diff
@@ -1993,6 +1993,136 @@
+        expect { giver.can_request_extension? }.to raise_error(SubmissionStatusGiver::AccessForbidden)
+      end
+    end
+
+    context "when status is 'collecting final submission files'" do
+      before { submission.status = 'collecting final submission files' }
+
+      it "raises an exception" do
+        giver = described_class.new(submission)
+        expect { giver.can_request_extension? }.to raise_error(SubmissionStatusGiver::AccessForbidden)
+      end
+    end
+
+    context "when status is 'waiting for advisor review'" do
+      before { submission.status = 'waiting for advisor review' }
+
+      it "raises an exception" do
+        giver = described_class.new(submission)
+        expect { giver.can_request_extension? }.to raise_error(SubmissionStatusGiver::AccessForbidden)
+      end
+    end
+
+    context "when status is 'waiting for committee review'" do
+      before { submission.status = 'waiting for committee review' }
+
+      it "raises an exception" do
+        giver = described_class.new(submission)
+        expect { giver.can_request_extension? }.to raise_error(SubmissionStatusGiver::AccessForbidden)
+      end
+    end
+
+    context "when status is 'waiting for head of program review'" do
+      before { submission.status = 'waiting for head of program review' }
+
+      it "raises an exception" do
+        giver = described_class.new(submission)
+        expect { giver.can_request_extension? }.to raise_error(SubmissionStatusGiver::AccessForbidden)
+      end
+    end
+
+    context "when status is 'waiting for final submission response'" do
+      before { submission.status = 'waiting for final submission response' }
+
+      it "raises an exception" do
+        giver = described_class.new(submission)
+        expect { giver.can_request_extension? }.to raise_error(SubmissionStatusGiver::AccessForbidden)
+      end
+    end
+
+    context "when status is 'waiting for publication release'" do
+      before { submission.status = 'waiting for publication release' }
+
+      it "raises an exception" do
+        giver = described_class.new(submission)
+        expect { giver.can_request_extension? }.to raise_error(SubmissionStatusGiver::AccessForbidden)
+      end
+    end
+
+    context "when status is 'waiting in final submission on hold'" do
+      before { submission.status = 'waiting in final submission on hold' }
+
+      it "raises an exception" do
+        giver = described_class.new(submission)
+        expect { giver.can_request_extension? }.to raise_error(SubmissionStatusGiver::AccessForbidden)
+      end
+    end
+
+    context "when status is 'released for publication'" do
+      before { submission.status = 'released for publication', submission.access_level = 'open_access' }
+
+      it "raises an exception" do
+        giver = described_class.new(submission)
+        expect { giver.can_request_extension? }.to raise_error(SubmissionStatusGiver::AccessForbidden)
+      end
```

Comment:
So I like this & it did improve the speed significantly. The only thing I realized is it makes it harder to tell which status is the problem if something does not go as expected. So it's speed/minimal code vs ease of troubleshooting. Thoughts?

Reply:
What about something along the lines of
`    [all submissions].each do |value, answer|
      it "throws an error when the status is #{submission.status}" do 
        giver = described_class.new(s) expect { giver.can_request_extension? }.to  raise_error(SubmissionStatusGiver::AccessForbidden)
      end
    end`

That way, there actually are separate tests for each status. (This may reduce the speed back to where it was, but would have both minimal code and troubleshooting.)

Reply:
This works & is in the middle speed wise!